### PR TITLE
Postgres: Operations, Hooks, OAuth login, Files support

### PR DESCRIPTION
**Pull request by flovilmart**
- Remove flaky tests on recursive roles load

### CODE REVIEW

**Comment by facebook-github-bot-0** [comment]
@flovilmart updated the pull request.

**Comment by facebook-github-bot-0** [comment]
@flovilmart updated the pull request.

**Comment by facebook-github-bot-0** [comment]
@flovilmart updated the pull request.

File: src/Adapters/Storage/Postgres/PostgresStorageAdapter.js [hunk 1]
```diff
@@ -412,7 +454,20 @@ export class PostgresStorageAdapter {
   // Drops a collection. Resolves with true if it was a Parse Schema (eg. _User, Custom, etc.)
   // and resolves with false if it wasn't (eg. a join table). Rejects if deletion was impossible.
   deleteClass(className) {
-    return notImplemented();
+    return this.getClass(className).then((schema) => {
+      let joins = joinTablesForSchema(schema);
```
Comment (drew-gross) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: Doesn't parse server drop the join tables automatically?

Comment (flovilmart) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: I'll try

Comment (flovilmart) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: I can confirm that can be simplified !

**Comment by facebook-github-bot-6** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2528/files/0d9c05b2b738b45c865317ab3e58a349cac04e9a..5112474c30d435831fa5583f9ca176158c334374)

File: .istanbul.yml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 instrumentation:
-  excludes: ["**/spec/**", "**/PostgresStorageAdapter.js"]
+  excludes: ["**/spec/**"]
```
Comment (flovilmart) [inline .istanbul.yml]: Yay let's start covering dat!

**Comment by facebook-github-bot-0** [comment]
@flovilmart updated the pull request.

**Comment by facebook-github-bot-0** [comment]
@flovilmart updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2528/files/bf81087178d9f45c79a0b48a053a115445d086ba..4da26f2c4521479272ee299841cc64516dda2600)

**Comment by facebook-github-bot-6** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2528/files/4da26f2c4521479272ee299841cc64516dda2600..b84fc4a3782aff69bc8f7de31c8dfa3089178f95)

**Comment by facebook-github-bot-1** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2528/files/b84fc4a3782aff69bc8f7de31c8dfa3089178f95..4da26f2c4521479272ee299841cc64516dda2600)

**Comment by facebook-github-bot-4** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2528/files/4da26f2c4521479272ee299841cc64516dda2600..7bb22b46a7e909c4f59990da556365403b3ac45b)

File: src/Adapters/Storage/Postgres/PostgresStorageAdapter.js [hunk 3]
```diff
@@ -349,22 +385,29 @@ export class PostgresStorageAdapter {
...
         // Table already exists, must have been created by a different request. Ignore error.
       } else {
         throw error;
       }
     }).then(() => {
       // Create the relation tables
       return Promise.all(relations.map((fieldName) => {
-        return this._client.none('CREATE TABLE IF NOT EXISTS $<joinTable:name> ("relatedId" varChar(120), "owningId" varChar(120), PRIMARY KEY("relatedId", "owningId") )', {joinTable: `_Join:${fieldName}:${className}`})
+        return this._client.none('CREATE TABLE IF NOT EXISTS $<joinTable:name> ("relatedId" varChar(120), "owningId" varChar(120), PRIMARY KEY("relatedId", "owningId") )', {joinTable: `_Join:${fieldName}:${className}`}).catch(err => {
+          console.error('ERROR CREATING JOIN!')
```
Comment (drew-gross) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: You sure we want to squelch this error?

Comment (flovilmart) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: Nope, I don't want to :p

**Comment by facebook-github-bot-1** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2528/files/be8644fef6dad7e4740faed3fdf060c6ff5cac5d..8bb493bc2dda065b05720b238bced7d098ea50d4)

**Comment by facebook-github-bot-7** [comment]
@flovilmart updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2528/files/8bb493bc2dda065b05720b238bced7d098ea50d4..7258d409c66ceab57543e5fda00f6b7f49e4d006)

**Pull request merged**